core/src/err/mod.rs

```diff
@@ -779,6 +779,10 @@ pub enum Error {
 	/// The db is running without an available storage engine
 	#[error("The db is running without an available storage engine")]
 	MissingStorageEngine,
+
+	/// Placeholder for internal methods to indicate unimplemented features
+	#[error("Placeholder for internal methods to indicate unimplemented features")]
+	UnimplementedInternally,
```

Comment:
There is already [an `Unimplemented` variant on this enum](https://docs.rs/surrealdb/1/surrealdb/error/enum.Db.html#variant.Unimplemented).

Reply:
Yes, but that requires a tuple of one `String` as associated data. That means I have to at least call `String::new()`, although technically speaking `String::new()` should not allocate yet because `Vec::new()`, the underlying call, should not allocate until needed as well.

Reply:
If you think that's okay, I'm fine with replacing `UnimplementedInternally` as `Unimplemented(String::new())`, but it might be confusing for others. Maybe a comment is needed to annotate why in this case.

Reply:
My worry is that we already have so many variants as it is and adding similar variants like this can cause confusion. Also if someone wants to check for unimplemented functionality they will have to check for the two variants now, instead of one. I think that variant embeds a string like that so that a more descriptive error message can be relayed.

Reply:
Okay, this should work.